### DISCUSSION TITLE
Comments to ignore links

### DISCUSSION THREAD

**Discussion by ehmicky** [lycheeverse/lychee#1447] [General] [answered]
Links can be ignored in many different ways, including with `--include`, `--exclude` and `.lycheeignore`. However, sometimes, it would be nice to ignore only one specific link in a given file.

A common way to do this is often to add a comment like `<!-- lychee-disable-next -->`, like [`markdown-link-check` does](https://github.com/tcort/markdown-link-check?tab=readme-ov-file#disable-comments), ESLint, and so on. 

One challenge is that those comments would be file format-specific. Other tools often opt to only implement this for Markdown, but there might be some creative solutions to make this format-agnostic.

I checked the issues and I do not think this has been asked before. What are your thoughts on this?

**Comment by mre** [comment, marked as answer]
As you said, we would need to handle various file formats, making the tool more complex and harder to maintain.
For instance, `markdown-link-check` also has edge-cases like `.mdx` support (https://github.com/tcort/markdown-link-check/issues/171) or anchor links (https://github.com/tcort/markdown-link-check/issues/196).

 `--exclude` and `.lycheeignore` should support that already, no? For example, instead of annotating a link in a file, why not add it to `.lycheeignore`?

**Reply by ehmicky** [reply]
That makes sense! Thanks for your response. :+1:

**Comment by mre** [comment]
Moved to a discussion, because this is technically not an issue with lychee. Hope that's fine.

**Reply by ehmicky** [reply]
That was more of a feature request, which many repositories handle with GitHub issues. 
I was following this part of [`CONTRIBUTING.md`](https://github.com/lycheeverse/lychee/blob/master/CONTRIBUTING.md#picking-an-issue):

> If you would like to contribute a new feature, the best way to get started is to ask for some early feedback by creating an issue yourself and asking for feedback.

Discussions work though! :+1:

**Reply by mre** [reply]
Ah yeah, makes sense. It's somewhat of a balancing act sometimes. I like to move feature requests, which need some prior discussion before they can be implemented to GitHub discussions. But yeah, I do this on a case-by-case basis.

**Comment by MaienM** [comment]
One way to avoid having to understand the different file formats would be to have a start and stop marker and simply ignore everything in-between. You would then not need to care what these comments/ignored sections look like, you'd just snip out the section between the two markers before processing the file. For example:

```html
<div>
	<a href="https://example.com/1">
	<!-- lychee-ignore-start -->
	<a href="https://example.com/2">
	<!-- lychee-ignore-end -->
	<a href="https://example.com/3">
</div>
```

would turn into

```html
<div>
	<a href="https://example.com/1">
	<!--  -->
	<a href="https://example.com/3">
</div>
```

and 

```typescript
doThing(
	'https://example.com/1',
	// lychee-ignore-start
	'https://example.com/2',
	// lychee-ignore-end
	'https://example.com/3',
)
```

would turn into

```typescript
doThing(
	'https://example.com/1',
	// 
	'https://example.com/3',
)
```

It's a bit more verbose than an ignore-next marker, but it avoids needing to understand what 'next' means.

Edit: I just noticed that this is essentially what was proposed in #1384, sorry for the noise, I had overlooked that one!
